pkg/placement/membership.go

```diff
@@ -307,9 +313,56 @@ func (p *Service) performTableDissemination() {
 func (p *Service) performTablesUpdate(hosts []placementGRPCStream, newTable *v1pb.PlacementTables) {
 	// TODO: error from disseminationOperation needs to be handle properly.
 	// Otherwise, each Dapr runtime will have inconsistent hashing table.
-	p.disseminateOperation(hosts, "lock", nil)
-	p.disseminateOperation(hosts, "update", newTable)
-	p.disseminateOperation(hosts, "unlock", nil)
+
+	startedAt := time.Now()
+
+	var wg sync.WaitGroup
+	for _, host := range hosts {
+		wg.Add(1)
+
+		go func(h placementGRPCStream) {
+			defer wg.Done()
+
+			timeoutC := make(chan bool)
+			stopC := make(chan bool)
+			step := ""
+			isTimeout := false
+
+			go func() {
+				defer close(stopC)
+
+				timeoutC <- true
+				step = "lock"
+				p.disseminateOperation([]placementGRPCStream{h}, "lock", nil)
+				if isTimeout {
+					return
+				}
+				step = "update"
+				p.disseminateOperation([]placementGRPCStream{h}, "update", newTable)
+				if isTimeout {
+					return
+				}
+				step = "unlock"
+				p.disseminateOperation([]placementGRPCStream{h}, "unlock", nil)
+			}()
+
+			<-timeoutC
+
+			select {
+			case <-time.After(10 * time.Second):
```

Comment:
quit host level 'lock -> update -> unlock' after 10s timeout, the goroutine will be left there, and the isTimeout flag will quit whatever next action asap.